encoders/firebase-decoders-json/src/json/java/com/google/firebase/decoders/json/ObjectDecoderContextImpl.java

```diff
@@ -47,6 +48,15 @@ private ObjectDecoderContextImpl(TypeToken.ClassToken<T> classToken) {
     }
   }
 
+  @NonNull
+  FieldDescriptor getFieldDescriptors(@NonNull String fieldName) {
+    if (fieldDescriptors.containsKey(fieldName)) {
+      return fieldDescriptors.get(fieldName);
+    } else {
+      throw new IllegalArgumentException(fieldName + " was not register in ObjectDecoder.");
```

Comment:
I don't think `IllegalArgumentException` is appropriate here, it usually indicates that the caller of the library provided some incorrect input. In this case it seems that the only time this could happen is if there is a bug in the library implementation, pls double check if that is the case and if so, change to `AssertionError`.

Reply:
Great Point here! I totally agree. it should never be null as long as the `fieldName` have associated `FieldRef`, in that case, it would make sense to make `FieldRef` as the key of the `fieldDescriptors`.